services/wallet/service.go

```diff
@@ -257,3 +261,29 @@ func (s *Service) Protocols() []p2p.Protocol {
 func (s *Service) IsStarted() bool {
 	return s.started
 }
+
+func (s *Service) SetKeycardPairingsFile(filePath string) {
+	s.keycardPairingsFile = filePath
+}
+
+func (s *Service) GetPairingsJSONFileContent() ([]byte, error) {
+	_, err := os.Stat(s.keycardPairingsFile)
+	if os.IsNotExist(err) {
+		return []byte{}, nil
+	}
+
+	return ioutil.ReadFile(s.keycardPairingsFile)
+}
+
+func (s *Service) SetPairingsJSONFileContent(content []byte) error {
+	_, err := os.Stat(s.keycardPairingsFile)
+	if os.IsNotExist(err) {
+		dir, _ := filepath.Split(s.keycardPairingsFile)
+		err = os.MkdirAll(dir, 0700)
+		if err != nil {
+			return err
+		}
+	}
+
+	return ioutil.WriteFile(s.keycardPairingsFile, content, 0600)
+}
```

Comment:
Service is entry point only to real logic, you can see how everything is being dispatch from the service. Could you move this into a keycard_paring_manager as we do elsewhere?

Reply:
@alaibe updated.